[BUG] Incorrect cast to int in `RemoteSegmentTransferTrackerTests.testStatsObjectCreationViaStream()` test

### Describe the bug

I am running into the following issue:

```sh
./gradlew ':server:test' --tests "org.opensearch.index.remote.RemoteSegmentTransferTrackerTests.testStatsObjectCreationViaStream" -Dtests.security.manager=true -Druntime.java=21
```

```sh
RemoteSegmentTransferTrackerTests > testStatsObjectCreationViaStream FAILED
    java.lang.AssertionError: expected:<2.147483647E9> but was:<5.037149882E9>
        at __randomizedtesting.SeedInfo.seed([17B4E7DE40BB6316:F74F7F9D73752E5]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:555)
        at org.junit.Assert.assertEquals(Assert.java:685)
        at org.opensearch.index.remote.RemoteSegmentTransferTrackerTests.testStatsObjectCreationViaStream(RemoteSegmentTransferTrackerTests.java:580)
```

The following line is causing the issue:

https://github.com/opensearch-project/OpenSearch/blob/13f04d97e3438b0389d1e851dfd08678ab644361/server/src/test/java/org/opensearch/index/remote/RemoteSegmentTransferTrackerTests.java#L580

When I add the following assert before the line 580
```java
assert deserializedStats.uploadTimeMovingAverage <= Integer.MAX_VALUE : "uploadTimeMovingAverage value too large for casting to int";
```
Then the assert fails

```sh
RemoteSegmentTransferTrackerTests > testStatsObjectCreationViaStream FAILED
    java.lang.AssertionError: uploadTimeMovingAverage value too large for casting to int
        at __randomizedtesting.SeedInfo.seed([C0053EEBF80305D6:D8C52ECC6F8F3425]:0)
        at org.opensearch.index.remote.RemoteSegmentTransferTrackerTests.testStatsObjectCreationViaStream(RemoteSegmentTransferTrackerTests.java:580)
```
What is the point of casting `double` to `int` when the original value can be larger than `Integer.MAX_VALUE` (for example `5037810928` in my case)?

In fact all the following three lines are IMO incorrectly using cast to `int`:

https://github.com/opensearch-project/OpenSearch/blob/13f04d97e3438b0389d1e851dfd08678ab644361/server/src/test/java/org/opensearch/index/remote/RemoteSegmentTransferTrackerTests.java#L574-L580

BTW, compare it to similar three lines from different test from the same class

https://github.com/opensearch-project/OpenSearch/blob/13f04d97e3438b0389d1e851dfd08678ab644361/server/src/test/java/org/opensearch/index/remote/RemoteSegmentTransferTrackerTests.java#L545-L547

I believe this issues is perhaps quick cut&paste issue or something along the lines.

This issue has been already partially reported in https://github.com/opensearch-project/OpenSearch/issues/10014

~~Feel free to assign this ticket to me, I will send PR.~~ PR sent.

### Related component

Storage:Remote

### To Reproduce

Please see the description above.


### Expected behavior

The test should pass.

### Additional Details

**Host/Environment (please complete the following information):**
 - OS: `% uname -a
Darwin Lukass-MBP.xxxx.xxxx 22.5.0 Darwin Kernel Version 22.5.0: Mon Apr 24 20:52:24 PDT 2023; root:xnu-8796.121.2~5/RELEASE_ARM64_T6000 arm64`
 - Version: `main` branch

--- a/server/src/test/java/org/opensearch/index/remote/RemoteSegmentTransferTrackerTests.java
+++ b/server/src/test/java/org/opensearch/index/remote/RemoteSegmentTransferTrackerTests.java
@@ -571,13 +571,9 @@ public class RemoteSegmentTransferTrackerTests extends OpenSearchTestCase {
                 assertEquals((int) deserializedStats.uploadBytesStarted, (int) transferTrackerStats.uploadBytesStarted);
                 assertEquals((int) deserializedStats.uploadBytesSucceeded, (int) transferTrackerStats.uploadBytesSucceeded);
                 assertEquals((int) deserializedStats.uploadBytesFailed, (int) transferTrackerStats.uploadBytesFailed);
-                assertEquals((int) deserializedStats.uploadBytesMovingAverage, transferTrackerStats.uploadBytesMovingAverage, 0);
-                assertEquals(
-                    (int) deserializedStats.uploadBytesPerSecMovingAverage,
-                    transferTrackerStats.uploadBytesPerSecMovingAverage,
-                    0
-                );
-                assertEquals((int) deserializedStats.uploadTimeMovingAverage, transferTrackerStats.uploadTimeMovingAverage, 0);
+                assertEquals(deserializedStats.uploadBytesMovingAverage, transferTrackerStats.uploadBytesMovingAverage, 0);
+                assertEquals(deserializedStats.uploadBytesPerSecMovingAverage, transferTrackerStats.uploadBytesPerSecMovingAverage, 0);
+                assertEquals(deserializedStats.uploadTimeMovingAverage, transferTrackerStats.uploadTimeMovingAverage, 0);
                 assertEquals((int) deserializedStats.totalUploadsStarted, (int) transferTrackerStats.totalUploadsStarted);
                 assertEquals((int) deserializedStats.totalUploadsSucceeded, (int) transferTrackerStats.totalUploadsSucceeded);
                 assertEquals((int) deserializedStats.totalUploadsFailed, (int) transferTrackerStats.totalUploadsFailed);

